frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -144,35 +151,51 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
     return null
   }
 
-  static getDeckObject = (props: PropsWithHeight): DeckObject => {
-    const { element, width, height, theme } = props
-    const json = JSON.parse(element.json)
+  static getDeckObject = (
+    props: PropsWithHeight,
+    state: Partial<State>
+  ): DeckObject => {
+    const { element, width, height, theme, isFullScreen } = props
+    const { id } = state
+
+    const currFullScreen = isFullScreen ?? false
+
+    // Only parse JSON when not transitioning to/from fullscreen, the element id changes, or theme changes
+    if (
+      element.id !== id ||
+      state.isFullScreen !== currFullScreen ||
+      state.isLightTheme !== hasLightBackgroundColor(theme)
```

Comment:
Why do we need to reparse this when layout changes? Aren't we always adapting the mapStyle below anyways based on the theme?